crates/crochet_infer/src/visitor.rs

```diff
@@ -1,60 +1,44 @@
 use crochet_ast::types::*;
 
 pub trait Visitor {
-    // fn visit_var(&mut self, tv: &mut TVar) {}
-    // fn visit_app(&mut self, app: &mut TApp) {}
-    // fn visit_lam(&mut self, lam: &mut TLam) {}
-    // fn visit_lit(&mut self, lit: &mut TLit) {}
-    // fn visit_keyword(&mut self, keyword: &mut TKeyword) {}
-    // fn visit_union(&mut self, union: &mut [Type]) {}
-    // fn visit_intersection(&mut self, intersection: &mut [Type]) {}
-    // fn visit_object(&mut self, object: &mut TObject) {}
-    // fn visit_ref(&mut self, r#ref: &mut TRef) {}
-    // fn visit_tuple(&mut self, tuple: &mut [Type]) {}
-    // fn visit_array(&mut self, array: &mut [Type]) {}
-    // fn visit_key_of(&mut self, key_of: &mut [Type]) {}
-    // fn visit_mutable(&mut self, mutable: &mut Type) {}
-    // fn visit_index_access(&mut self, index_access: &mut TIndexAccess) {}
-    // fn visit_generic(&mut self, generic: &mut TGeneric) {}
-
     fn visit_type(&mut self, t: &mut Type);
 
     fn visit_children(&mut self, t: &mut Type) {
-        match t.kind {
+        match &mut t.kind {
```

Comment:
This is equivalent to the old code and follows what we do in the rest of the codebase.